tests/unit_tests.py

```diff
@@ -37,11 +37,11 @@
 os.chdir(os.environ['APP_DIR'])
 
 # ...but we want pytest to default to finding tests in TESTS_DIR, not the cwd.
-# So, when no arguments are given, we make the option parser return TESTS_DIR.
-import _pytest.config
-original_parse_setoption = _pytest.config.Parser.parse_setoption
-_pytest.config.Parser.parse_setoption = (lambda *args, **kwargs: 
-    original_parse_setoption(*args, **kwargs) or [os.environ['TESTS_DIR']])
+# So, when no arguments besides options are given, we add TESTS_DIR to the list
+# of options passed to pytest.
+args = sys.argv[1:]
+if all([arg.startswith('-') for arg in args]):
```

Comment:
One typical usage of tools/unit_test is e.g.:

    $ tools/unit_test -k foo

to filter tests. This doesn't pass this check I believe because `foo` doesn't start with `-`. So we do need to call the parser unfortunately?

Reply:
Is there something special about the letter k?

With this commit, that wouldn't work, but you can run a specific test by just passing it as a regular argument, e.g.:
`$ tools/unit_tests test_pfif`
or you can run multiple like this:
`$ tools/unit_tests test_pfif test_photo`

Reply:
Oh I didn't know that I can pass test names as regular arguments. But when you write:
```$ tools/unit_tests test_pfif```
doesn't it also prevent it from adding TESTS_DIR to `args`?

Here is document of -k option (run `pytest -h` to get this). I guess it's more flexible than regular arguments?:
```
  -k EXPRESSION         only run tests which match the given substring
                        expression. An expression is a python evaluatable
                        expression where all names are substring-matched
                        against test names and their parent classes. Example:
                        -k 'test_method or test_other' matches all test
                        functions and classes whose name contains
                        'test_method' or 'test_other', while -k 'not
                        test_method' matches those that don't contain
                        'test_method' in their names. Additionally keywords
                        are matched to classes and functions containing extra
                        names in their 'extra_keyword_matches' set, as well as
                        functions which have names assigned directly to them.
```

Reply:
Ah, I see what you're saying. Yeah, this isn't going to handle it correctly. I've just gone back to what it was doing before with the parser and switched it to using the new package. I did it on my other branch (pull request #423). I'll close this pull request now.